x/wasm/types/authz_test.go

```diff
@@ -744,3 +745,184 @@ func mustGrant(contract sdk.AccAddress, limit ContractAuthzLimitX, filter Contra
 	}
 	return *g
 }
+
+func TestValidateCodeGrant(t *testing.T) {
+	specs := map[string]struct {
+		codeHash              []byte
+		instantiatePermission AccessConfig
+		expErr                bool
+	}{
+		"all good": {
+			codeHash:              []byte("ABC"),
+			instantiatePermission: AllowEverybody,
+		},
+		"empty code hash": {
+			codeHash:              []byte{},
+			instantiatePermission: AllowEverybody,
+			expErr:                true,
+		},
+		"nil code hash": {
+			codeHash:              nil,
+			instantiatePermission: AllowEverybody,
+			expErr:                true,
+		},
+		"invalid permission": {
+			codeHash:              []byte("ABC"),
+			instantiatePermission: AccessConfig{Permission: AccessTypeUnspecified},
+			expErr:                true,
+		},
```

Comment:
good tests. Please add a case with empty permission so that we have this covered for regression.